### PR TITLE
feat: Show link location in /bookings

### LINKED ISSUE #11730: [CAL-2590] Show link location in /bookings
show link to join a call in the /booking overview

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2590](https://linear.app/calcom/issue/CAL-2590/show-link-location-in-bookings)</sub>

### PR DESCRIPTION
**Pull request by kremedev**
## What does this PR do?

Fixes #11730

/claim #11730

https://github.com/calcom/cal.com/assets/134702704/f288c36d-6ad1-465c-8e6c-7f5ac107121c

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/components/booking/BookingListItem.tsx [hunk 1]
```diff
@@ -353,6 +353,28 @@ function BookingListItem(booking: BookingItemProps) {
...
+                        height={16}
+                        width={16}
+                      />
+                      Join Google Meet
+                    </div>
+                  ) : (
+                    "Meeting Link"
+                  )}
+                </Link>
+              )}
```
Comment (PeerRich) [inline apps/web/components/booking/BookingListItem.tsx:377]: we should not hard code logos and names, instead use whatever conference location was used

**Review by PeerRich — DISMISSED** [review]
we should not hard code logos and names, instead use whatever conference location was used

**Comment by kremedev** [comment]
I tried using location link inside booking prop to get app logo and name but some apps like Google and Zoom use this type.
```type DynamicLinkBasedEventLocation = {
  linkType: "dynamic";
  urlRegExp?: null;
  organizerInputType?: null;
  organizerInputPlaceholder?: null;
} & CommonProperties;
```
Should I change this type and add urlRegExp to them?

**Comment by Siddharth-2382** [comment]
Umm I tried this code and I don't think `booking.location` is giving you the meeting link. It just gives value like `integrations:zoom` or `integrations:google:meet`. Can you share a recording? I'm bit lost here.

**Comment by kremedev** [comment]
I updated the recording

**Comment by Siddharth-2382** [comment]
> I updated the recording

When I'm clicking on the link it takes to /integrations:google%20meet?

**Comment by kremedev** [comment]
It shouldn't work at cal, daily, huddle, jets, and tandem since I haven't set them up yet.

I don't know how you get `integrations:google:meet` from `booking.location`. Is this how you created it?

Ekran Resmi 2023-10-13 21 03 04

**Comment by Siddharth-2382** [comment]
> It shouldn't work at cal, daily, huddle, jets, and tandem since I haven't set them up yet.
> 
> I don't know how you get `integrations:google:meet` from `booking.location`. Is this how you created it?
> 
> Ekran Resmi 2023-10-13 21 03 04

No I simply created an event with location set as google meet.

**Comment by kremedev** [comment]
> > It shouldn't work at cal, daily, huddle, jets, and tandem since I haven't set them up yet.
> > I don't know how you get `integrations:google:meet` from `booking.location`. Is this how you created it?
> > Ekran Resmi 2023-10-13 21 03 04
> 
> No I simply created an event with location set as google meet. Can you show by going to the google meet link that you're having? Like a screen recording?


https://github.com/calcom/cal.com/assets/134702704/55e09422-3b16-4c0d-8d36-545d114e5474

File: apps/web/components/booking/BookingListItem.tsx [hunk 2]
```diff
@@ -275,6 +275,56 @@ function BookingListItem(booking: BookingItemProps) {
...
+              href={link}
+              onClick={(e) => e.stopPropagation()}
+              className="flex items-center gap-1 text-sm leading-6 text-blue-400 hover:underline">
+              <img src={app.iconUrl} alt={app.label} height={16} width={16} />
+              Join {app.label}
+            </Link>
+          </div>
+        );
+      }
+      const url = booking.references[0].meetingUrl;
```
Comment (kremedev) [inline apps/web/components/booking/BookingListItem.tsx:301]: Is this the correct way to get urls from integrations @PeerRich ?

Comment (PeerRich) [reply apps/web/components/booking/BookingListItem.tsx:301]: i believe so. asking @hariombalhara to review

Comment (PeerRich) [reply apps/web/components/booking/BookingListItem.tsx:301]: https://github.com/calcom/cal.com/pull/11866 this looks better approach

**Comment by PeerRich** [comment]
https://github.com/calcom/cal.com/pull/11866 going with this one

File: apps/web/components/booking/BookingListItem.tsx (hunk 2 above)
Comment (kremedev) [reply apps/web/components/booking/BookingListItem.tsx:301]: I didn't know you wanted Join Meeting link to go booking page

**Pull request closed without merging**